src/Microsoft.DotNet.CodeFormatting/FormattingEngineImplementation.cs

```diff
@@ -179,10 +180,18 @@ public ImmutableArray<DiagnosticDescriptor> AllSupportedDiagnostics
 
         public async Task FormatSolutionWithAnalyzersAsync(Solution solution, CancellationToken cancellationToken)
         {
-            foreach (var project in solution.Projects)
+            var watch = new Stopwatch();
+            watch.Start();
+
+            var workspace = solution.Workspace;
+            foreach (var projectId in solution.ProjectIds)
             {
+                var project = workspace.CurrentSolution.GetProject(projectId);
```

Comment:
The Format method applies a codefix which applies changes to the workspace. That means that the currentsolution of the workspace would get updated. So unless we get the project from the current solution, we'd be formatting a Project from a previous solution and Workspace.ApplyChanges would reject the changes to that Project (because it's not to the latest snapshot). The net effect is that without this change only the first project would get formatted.